pkg/agentfunctions/provide.go

```diff
@@ -32,8 +32,11 @@ var Module = fx.Options(
 // +kubebuilder:object:generate=true
 type Config struct {
 	// RPC servers to connect to (which will be able to call agent functions)
+	//+kubebuilder:validation:Optional
```

Comment:
This can be made global like our other packages -- e.g. on top of pkg/config/config.go I had added --

```
// +kubebuilder:validation:Optional
package config
```

Reply:
cool! That's certainly better idea.

Btw, as you've improved swagger generation recently, we should ditch kubebuilder generation (at some point). Right now we have basically two json-schema generators which is a bit weird

Reply:
(ok, we can't do that, one is for proto and one for golang stuff)